inject-java/src/test/groovy/io/micronaut/aop/targetaware/TypeSpecificTargetAwareInterceptor.java

```diff
@@ -0,0 +1,35 @@
+package io.micronaut.aop.targetaware;
+
+import io.micronaut.aop.MethodInvocationContext;
+import io.micronaut.aop.TargetAwareMethodInterceptor;
+import io.micronaut.context.Qualifier;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.spockframework.util.Assert;
+
+import javax.inject.Singleton;
+
+@Singleton
+public class TypeSpecificTargetAwareInterceptor implements TargetAwareMethodInterceptor<TestBean, Object> {
+    private TestBean target;
+    private int count;
+    @Override
+    public Object intercept(MethodInvocationContext<TestBean, Object> context) {
+        Assert.notNull(target);
+        return context.proceed();
+    }
+
+    @Override
+    public void newTarget(@Nullable Qualifier<TestBean> qualifier, @NotNull TestBean target) {
+        count++;
```

Comment:
Nice. Yes, this would be helpful for initialization of [proto and micro fields](https://github.com/jtulach/micronautui/blob/04ff91230ffa470536e7c0a88d2d3d20bfa3ab5a/beaninfo-api/src/main/java/com/dukescript/impl/javafx/beans/ObservableInterceptor.java#L22) as soon as the `ObservableInterceptor` instance is created.